CHANGELOG.md

```diff
@@ -1,4 +1,144 @@
 ## Solidus 3.1.0 (master, unreleased)
+
+### Major changes
+
+**`Spree.load_defaults`: preference defaults depending on the Solidus version**
+
+Solidus 3.1 brings a new feature where preference defaults can take different
+values depending on a specified Solidus version. It makes it possible to stop
+deprecating old defaults every time we introduce a change in the recommended
+value for a setting. After all, they're just that; recommendations. Instead,
+now users can explicitly ask for a given Solidus version defaults and, as
+before, override the preferences they want.
+
+When upgrading to 3.1, you have to take action to adopt the new behavior.
+You'll need to add `Spree.load_defaults('3.1')` on the very top of your
+`spree.rb` initializer. As we're not changing any preference default on this
+release, nothing will break. A warning will be emitted on boot-up until you do
+it!
+
+However, bumping the version given to `load_defaults` straight away for future
+upgrades will not be a safe option. Instead, you'll have to go through the new
+update process detailed below.
+
+- Allow using different preference defaults depending on a Solidus version [#4064](https://github.com/solidusio/solidus/pull/4064) ([waiting-for-dev](https://github.com/waiting-for-dev))
+
+**New update process**
+
+As aforementioned, preference defaults can change after a Solidus release. Once
+you have your defaults locked to the current Solidus version, a new upgrade
+won't break your application because of them. However, it's a good idea to
+adapt your application to the updated recommended settings. To help with this
+process, Solidus comes with a generator that you can execute like this:
+
+```bash
+bin/rails g solidus:update
+```
+
+That generator will create a new initializer called `new_solidus_defaults.rb`,
+which will preview all the defaults that have changed between versions, each on
+a commented line. From that point, you can activate the new defaults one by one
+and adapt your application incrementally. Once you're done with all of them,
+you can bump the version given to `Spree.load_defaults` in the `spree.rb`
+initializer and remove the `new_solidus_defaults.rb` initializer altogether.
+
+You can read in more detail about [this process on our
+guides](https://edgeguides.solidus.io/getting-started/upgrading-solidus#updating-preferences).
+
+- Introduce Solidus update process [#4087](https://github.com/solidusio/solidus/pull/4087) ([waiting-for-dev](https://github.com/waiting-for-dev))
+
+**Other important changes**
+
+`Spree::Price#amount` field can no longer be `nil`. Besides adding the
+validation at the model layer, we ship with a task that will remove records
+where the amount is `NULL` in the database. You should run the task before
+executing the new migrations:
```

Comment:
I see some problems here when deploying these changes to staging/production. The script should be run before migrations but usually migrations are executed automatically on deploy. I think the only way to solve this is by suggesting people to copy/paste the rake task in their application and run it before deploying the upgrade. Do you any other ideas?

Reply:
That's definitely a good point. Yes, they could copy it into their own task or enter the console and manually executing it (which is always dangerous in production). I see no other way. Because we could add the task into the database migration itself, but I don't think it's safe to delete user records blindly without making them knowledgeable of it.

I added a new commit with the additional notes. Please, take a look. The link to the task won't work because it's waiting for #4157 to be merged.